### PR TITLE
feat: add integrity/validation check for tarball/manifest diff

### LINKED ISSUE #145: Compare and validate packument and tarball manifest
The scanner should search for obvious tentative of confusion between the packument and the manifest in the tarball.

See/read: https://blog.vlt.sh/blog/the-massive-hole-in-the-npm-ecosystem

### PR DESCRIPTION
**Pull request by fraxken**
This PR fixes #145

If a package tarball and manifest integrity doesn't match then we add a global warning:
image

What do we compare? (in the same order)
- name
- version
- dependencies (or empty Object if no value)
- license (or empty string if no value)
- scripts (or empty Object is no value)

> Please feel free to notify others that we may absolutely add

### CODE REVIEW

**Comment by fraxken** [comment]
TODO:

- We need to validate the behavior for a tree with multiple versions of the same packages!
- Add FT with https://www.npmjs.com/package/darcyclarke-manifest-pkg

File: test/integrityWarning.spec.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Require Node.js Dependencies
+import { test } from "node:test";
+import assert from "node:assert";
+
+// Require Internal Dependencies
+import { from } from "../index.js";
```
Comment (PierreDemailly) [inline test/integrityWarning.spec.js:6]:
```suggestion
// Import Node.js Dependencies
import { test } from "node:test";
import assert from "node:assert";

// Import Internal Dependencies
import { from } from "../index.js";
```

**Review by PierreDemailly — APPROVED** [review]

**Review by PierreDemailly — APPROVED** [review]

File: test/manifest.spec.js [hunk 2]
```diff
@@ -101,3 +102,36 @@ test("manifest.readAnalyze should return hasNativeElements: true because of the
...
+    },
+    license: "MIT",
+    scripts: {
+      test: "hello world!"
+    }
+  };
+  const expectedIntegrity = crypto
+    .createHash("sha256")
+    .update(JSON.stringify(manifestJSON))
+    .digest("hex");
```
Comment (tony-go) [inline test/manifest.spec.js:122]: Maybe we should expose (not to the public API) a `generateIntegrityHash` method we could reuse here?

Comment (fraxken) [reply test/manifest.spec.js:122]: Didn't abstract it cause only used it two times. Also input and context in both cases were not exactly the same.

And having a function just to abstract crypto.createHash does not feel useful to me.

**Pull request merged**